csharp/dotnet/security/use_ecb_mode.yaml

```diff
@@ -0,0 +1,28 @@
+rules:
+  - id: use_ecb_mode
+    patterns:
+      - pattern-either:
+          - pattern: ($KEYTYPE $KEY).EncryptEcb(...);
+          - pattern: ($KEYTYPE $KEY).DecryptEcb(...);
+          - pattern: ($KEYTYPE $KEY).Mode = CipherMode.ECB;
+      - metavariable-pattern:
+          metavariable: $KEYTYPE
+          pattern-either:
+            - pattern: SymmetricAlgorithm
+            - pattern: Aes
+            - pattern: Rijndael
+            - pattern: DES
+            - pattern: TripleDES
+            - pattern: RC2
+    message: Usage of ECB mode detected
```

Comment:
Could you expand the message to also include a recommendation of what to use instead?
You could, for example, reuse or rephrase this section from the OWASP cheatsheet:

> Where available, authenticated modes should always be used. These provide guarantees of the integrity and authenticity of the data, as well as confidentiality. The most commonly used authenticated modes are [GCM](https://en.wikipedia.org/wiki/Galois/Counter_Mode) and [CCM](https://en.wikipedia.org/wiki/CCM_mode), which should be used as a first preference.
> 
> If GCM or CCM are not available, then [CTR](https://en.wikipedia.org/wiki/Block_cipher_mode_of_operation#Counter_%28CTR%29) mode or [CBC](https://en.wikipedia.org/wiki/Block_cipher_mode_of_operation#Cipher_Block_Chaining_%28CBC%29) mode should be used. As these do not provide any guarantees about the authenticity of the data, separate authentication should be implemented,

https://cheatsheetseries.owasp.org/cheatsheets/Cryptographic_Storage_Cheat_Sheet.html#cipher-modes

Reply:
a76330075690b0e2bf93264a5762b6c3436da5a5